lib/puppet/provider/dns_record/dynect.rb

```diff
@@ -0,0 +1,149 @@
+
+# Author:: Charles Dunbar <charles@puppetlabs.com>
+# Type Name:: dns_record
+# Provider:: dynect
+#
+# Copyright 2015, Puppet Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'json'
```

Comment:
Yeah. You're right. In this case it wouldn't apply, but in general they should go inside the provider block.